src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -2529,23 +2529,81 @@ public static ArgumentSyntax Argument(ExpressionSyntax expression)
         /// <summary>Creates a new ArgumentSyntax instance.</summary>
         public static ArgumentSyntax Argument(NameColonSyntax nameColon, SyntaxToken outKeyword, VariableDeclarationSyntax declaration)
         {
-            if (outKeyword.Kind() != SyntaxKind.OutKeyword)
+            return Argument(nameColon, outKeyword, (CSharpSyntaxNode)declaration);
```

Comment:
Looking at the method's signature a couple of lines above, I notice that it is still public.
But in your description, you wrote that it will be internal:
> Factory static ArgumentSyntax Argument(NameColonSyntax nameColon, SyntaxToken refOrOutKeyword, CSharpSyntaxNode expressionOrDeclaration) will be internal.